### PR TITLE
DEV-3015: Mint views

### PR DESCRIPTION
**Pull request by johanventer**
## Description of the change

> Mint components initial 1.0 deploy.

Support for the Service Titan portal has been deprecated as of mint-components version `1.1.x`. Please see the comments in the `styles.ts` file for how the Service Titan portal should be updated to support mint-components version `1.1.x+`.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3015
 - Process.st launch checklist: https://app.process.st/checklists/Mint-Components-10-ltJCOF78YZ9Bo7LVtIpLuA

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: packages/component-boilerplate/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/component-boilerplate",
-  "version": "1.0.0",
+  "version": "1.0.1-20",
   "private": false,
```
Comment (loganvolkers) [inline packages/component-boilerplate/package.json:4]: should not have any `component-boilerplate` changes in this branch

File: packages/mint-components/package.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@saasquatch/mint-components",
+  "version": "1.0.0-70",
```
Comment (loganvolkers) [inline packages/mint-components/package.json:3]: Should be a 1.0 main release

```suggestion
  "version": "1.0.0",
```

File: packages/mint-components/src/components/sqm-navigation-sidebar-item/sqm-navigation-sidebar-item-view.tsx [hunk 3]
```diff
@@ -47,7 +48,11 @@ export function NavigationSidebarItemView(
   const styleString = sheet.toString();
 
   return (
-    <div onClick={callbacks.onClick} class={sheet.classes.ItemContainer}>
+    <div
+      onClick={callbacks.onClick}
+      onMouseDown={(e) => (e.button === 1 ? callbacks.onMiddleClick(e) : true)}
+      class={sheet.classes.ItemContainer}
+    >
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-navigation-sidebar-item/sqm-navigation-sidebar-item-view.tsx:55]: These should just be simple `<a href="/wherever">` components, without all the special `onMouseDown` handling.

File: packages/mint-components/src/components/sqm-share-code/useShareCode.tsx [hunk 4]
```diff
@@ -24,11 +25,24 @@ const MessageLinkQuery = gql`
 `;
 
 export function useShareCode(props: ShareCodeProps): ShareLinkViewProps {
-  const { programId = useProgramId() } = props;
+  const programId = useDomContext("sq:program-id");
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-share-code/useShareCode.tsx]: Hmm, why can't this `useProgramId`? Why does it have to use the context directly?

File: packages/mint-components/tsconfig.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
     "moduleResolution": "node",
     "module": "esnext",
     "target": "es2019",
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Comment (johanventer) [inline packages/mint-components/tsconfig.json]: Why?

File: packages/mint-components/src/components/sqm-share-code/useShareCode.tsx (hunk 4 above)
Comment (00salmon) [reply packages/mint-components/src/components/sqm-share-code/useShareCode.tsx]: this is incomplete 🙈

File: packages/mint-components/tsconfig.json (hunk 5 above)
Comment (00salmon) [reply packages/mint-components/tsconfig.json]: I'm not sure when this was added... i'm guessing it was a shortcut for demo hooks where the props are usually unused

File: packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx [hunk 6]
```diff
@@ -35,6 +35,7 @@ export function DividedLayoutView(
     display: flex;
     flex: 1;
     flex-direction: ${props.direction};
+    background-color: var(--st-portal-background);
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx:38]: This is `mint-components`, should we be using `st-*` variables here?

File: packages/mint-components/src/components/sqm-portal-register/sqm-portal-register-view.tsx [hunk 7]
```diff
@@ -65,6 +69,8 @@ export function PortalRegisterView(props: PortalRegisterViewProps) {
     window.scrollTo({ top: 0, left: 0, behavior: "smooth" });
   }
 
+  console.log("view", states, states.validationState);
+
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-portal-register/sqm-portal-register-view.tsx:73]: Flagging console.log

File: packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.tsx [hunk 8]
```diff
@@ -72,6 +72,8 @@ export class PortalRegister {
       submitLabel: this.submitLabel,
       pageLabel: this.pageLabel,
     };
+
+    console.log({states})
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.tsx:76]: Flagging console.log

File: packages/mint-components/src/components/sqm-portal-register/usePortalRegister.tsx [hunk 9]
```diff
@@ -1,15 +1,17 @@
...
 } from "@saasquatch/component-boilerplate";
+import { useValidationState } from "./useValidationState";
 
 export function usePortalRegister({ nextPage, confirmPassword }) {
-  const [error, setError] = useState("");
+  const { validationState, setValidationState } = useValidationState({});
   const [request, { loading, errors, data }] =
     useRegisterWithEmailAndPasswordMutation();
 
+  console.log({ validationState });
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-portal-register/usePortalRegister.tsx:14]: Flagging console.log

File: packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx [hunk 10]
```diff
@@ -35,7 +35,7 @@ export function DividedLayoutView(
     display: flex;
     flex: 1;
     flex-direction: ${props.direction};
-    background-color: var(--st-portal-background);
+    background-color: var(--st-content-background);
```
Comment (johanventer) [inline packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx:38]: This is `mint-components`, should we be using `st-*` variables here?

Comment (ianhitchcock) [reply packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx]: replaced with -sqm

Comment (noahwc) [reply packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx]: When we were doing service titan some service titan styles ended up leaking into mint components. We should probably update the service titan project to have these styles and remove them from mint.

Comment (noahwc) [reply packages/mint-components/src/components/sqm-divided-layout/sqm-divided-layout-view.tsx]: Going to deprecate ST support in the current mint components version. I'll add the breaking changes to the changelog. Should be an easy fix on the ST end if we ever want to update them.

**Review by DerekSiemens — APPROVED** [review]

**Review by DerekSiemens — APPROVED** [review]

**Pull request merged**